File: src/wsTripXML/Code/OTAClasses/OTACruiseBooking.vb
Option Strict Off
Option Explicit On 

Imports System.Xml.Serialization

Namespace wsTravelTalk.wmCruiseBooking

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class CompanyName

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CompanyShortName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public TravelSector As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Code As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CodeContext As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedCategory

        '<remarks/>
        <System.Xml.Serialization.XmlArrayItemAttribute(IsNullable:=False)> _
        Public CabinFilters() As CabinFilter

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("SelectedCabin")> _
        Public SelectedCabin() As SelectedCabin

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public BerthedCategoryCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PricedCategoryCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DeckName As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedCabin

        '<remarks/>
        <System.Xml.Serialization.XmlArrayItemAttribute(IsNullable:=False)> _
        Public CabinFilters() As CabinFilter

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Status As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ReleaseDateTime As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CabinNumber As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public BedConfigurationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public MaxOccupancy As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DeclineIndicator As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public HeldIndicator As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class CabinFilter

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CabinFilterCode As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class CabinFilters

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("CabinFilter")> _
        Public CabinFilter() As CabinFilter
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class Currency

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CurrencyCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DecimalPlaces As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class InclusivePackageOption

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CruisePackageCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public InclusiveIndicator As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public StartDate As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedSailing

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VoyageID As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Start As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Duration As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public [End] As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VendorCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VendorName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ShipCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ShipName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(DataType:="NMTOKEN")> _
        Public Status As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class GuestCity

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public LocationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CodeContext As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class GuestTransportation

        '<remarks/>
        Public GuestCity As GuestCity

        '<remarks/>
        Public GatewayCity As GatewayCity

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public TransportationMode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(DataType:="NMTOKEN")> _
        Public TransportationStatus As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class GatewayCity

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public LocationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CodeContext As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class ListItem

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Formatted As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Language As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute("ListItem")> _
        Public ListItem1 As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class [Text]

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Formatted As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Language As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

End Namespace

File: src/wsTripXML/Code/OTAClasses/OTACruiseBooking.cs

using System.Xml.Serialization;

namespace wsTripXML.wsTravelTalk.wmCruiseBooking
{

    // local CompanyName removed: identical to shared wsTripXML.Code.CompanyName (XML type-name collision in one serializer scope)

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedCategory
    {

        // <remarks/>
        [XmlArrayItem(IsNullable = false)]
        public CabinFilter[] CabinFilters;

        // <remarks/>
        [XmlElement("SelectedCabin")]
        public SelectedCabin[] SelectedCabin;

        // <remarks/>
        [XmlAttribute()]
        public string BerthedCategoryCode;

        // <remarks/>
        [XmlAttribute()]
        public string PricedCategoryCode;

        // <remarks/>
        [XmlAttribute()]
        public string DeckName;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedCabin
    {

        // <remarks/>
        [XmlArrayItem(IsNullable = false)]
        public CabinFilter[] CabinFilters;

        // <remarks/>
        [XmlAttribute()]
        public string Status;

        // <remarks/>
        [XmlAttribute()]
        public string ReleaseDateTime;

        // <remarks/>
        [XmlAttribute()]
        public string CabinNumber;

        // <remarks/>
        [XmlAttribute()]
        public string BedConfigurationCode;

        // <remarks/>
        [XmlAttribute()]
        public string MaxOccupancy;

        // <remarks/>
        [XmlAttribute()]
        public string DeclineIndicator;

        // <remarks/>
        [XmlAttribute()]
        public string HeldIndicator;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class CabinFilter
    {

        // <remarks/>
        [XmlAttribute()]
        public string CabinFilterCode;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class CabinFilters
    {

        // <remarks/>
        [XmlElement("CabinFilter")]
        public CabinFilter[] CabinFilter;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Currency
    {

        // <remarks/>
        [XmlAttribute()]
        public string CurrencyCode;

        // <remarks/>
        [XmlAttribute()]
        public string DecimalPlaces;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class InclusivePackageOption
    {

        // <remarks/>
        [XmlAttribute()]
        public string CruisePackageCode;

        // <remarks/>
        [XmlAttribute()]
        public string InclusiveIndicator;

        // <remarks/>
        [XmlAttribute()]
        public string StartDate;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedSailing
    {

        // <remarks/>
        [XmlAttribute()]
        public string VoyageID;

        // <remarks/>
        [XmlAttribute()]
        public string Start;

        // <remarks/>
        [XmlAttribute()]
        public string Duration;

        // <remarks/>
        [XmlAttribute()]
        public string End;

        // <remarks/>
        [XmlAttribute()]
        public string VendorCode;

        // <remarks/>
        [XmlAttribute()]
        public string VendorName;

        // <remarks/>
        [XmlAttribute()]
        public string ShipCode;

        // <remarks/>
        [XmlAttribute()]
        public string ShipName;

        // <remarks/>
        [XmlAttribute(DataType = "NMTOKEN")]
        public string Status;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class GuestCity
    {

        // <remarks/>
        [XmlAttribute()]
        public string LocationCode;

        // <remarks/>
        [XmlAttribute()]
        public string CodeContext;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class GuestTransportation
    {

        // <remarks/>
        public GuestCity GuestCity;

        // <remarks/>
        public GatewayCity GatewayCity;

        // <remarks/>
        [XmlAttribute()]
        public string TransportationMode;

        // <remarks/>
        [XmlAttribute(DataType = "NMTOKEN")]
        public string TransportationStatus;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class GatewayCity
    {

        // <remarks/>
        [XmlAttribute()]
        public string LocationCode;

        // <remarks/>
        [XmlAttribute()]
        public string CodeContext;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class ListItem
    {

        // <remarks/>
        [XmlAttribute()]
        public string Formatted;

        // <remarks/>
        [XmlAttribute()]
        public string Language;

        // <remarks/>
        [XmlAttribute("ListItem")]
        public string ListItem1;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Text
    {

        // <remarks/>
        [XmlAttribute()]
        public string Formatted;

        // <remarks/>
        [XmlAttribute()]
        public string Language;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

}